src/packet.rs

```diff
@@ -12,6 +12,18 @@ pub use self::outgoing::{Outgoing, OutgoingPacket, OutgoingPacketBuilder};
 pub use self::packet_reader::PacketReader;
 pub use self::packet_structure::Packet;
 
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum ItemIdentifier {
```

Comment:
Erm.. Couldn't this be an `Option<SequenceNumber>` instead of a new type?

Reply:
It makes it hard to see at call-sites what you're sending in, especially if you have two
```
call(
    ...,
    Some(1),
    Some(1),
)
```

Reply:
Perhaps my biggest complaint is that it isn't a great name. It's more of a MessageId than anything.

Reply:
Also the type names are are verbs? I don't know. I don't love it. An option would do just fine here

Reply:
We use item_identifier in the code to represent the item Id. I really prefer to use a separate type instead of an Option. An option is also a weaker type, we can accidentally mix two arguments wrongly.

For the verb part, it's because they're just used locally to store values, and tell the sequencer if we want a new value or use an existing value. They can be changed though. What do you propose?

Reply:
If you really want to be that pedantic about types, technically there should then be a type alias for ItemIdentifier since it is not technically a sequence number. Then you can call it OptionItemIdentifier with Some and None varients as a new enum. I really think an option connotation is what we want here.